driver-core/src/main/java/com/datastax/driver/core/CodecRegistry.java

```diff
@@ -279,32 +302,50 @@ public void onRemoval(RemovalNotification<CacheKey, TypeCodec<?>> notification)
      */
     public CodecRegistry() {
         this.codecs = new CopyOnWriteArrayList<TypeCodec<?>>(PRIMITIVE_CODECS);
-        this.cache = CacheBuilder.newBuilder()
+        this.cache = defaultCacheBuilder().build(new TypeCodecCacheLoader());
+    }
+
+    private CacheBuilder<CacheKey, TypeCodec<?>> defaultCacheBuilder() {
+        CacheBuilder<CacheKey, TypeCodec<?>> builder = CacheBuilder.newBuilder()
             // 19 primitive codecs + collections thereof = 19*3 + 19*19 = 418 codecs,
             // so let's start with roughly 1/4 of that
             .initialCapacity(100)
             .weigher(new TypeCodecWeigher())
-            // a cache with all 418 "basic" codecs weighs 399 (19*2 + 19*19),
-            // so let's cap at roughly 2.5x this size
-            .maximumWeight(1000)
-            .concurrencyLevel(Runtime.getRuntime().availableProcessors() * 4)
-            .removalListener(new TypeCodecRemovalListener())
-            .build(
-                new CacheLoader<CacheKey, TypeCodec<?>>() {
-                    public TypeCodec<?> load(CacheKey cacheKey) {
-                        return findCodec(cacheKey.cqlType, cacheKey.javaType);
-                    }
-                });
+            .maximumWeight(1000);
+        if (logger.isTraceEnabled())
+            // do not bother adding a listener if it will be ineffective
+            builder = builder.removalListener(new TypeCodecRemovalListener());
```

Comment:
It's possible that log levels can be manipulated dynamically, but that seems like such a thin use case that we don't need to worrya bout it.